pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -78,13 +84,53 @@ func (r *PullSecretReconciler) Reconcile(request ctrl.Request) (ctrl.Result, err
 			ps.Data = map[string][]byte{}
 		}
 
-		// validate
-		if !json.Valid(ps.Data[v1.DockerConfigJsonKey]) {
+		keyCondition := &status.Condition{
+			Type:   aro.RedHatKeyPresent,
+			Status: v1.ConditionFalse,
+			Reason: "CheckFailed",
+		}
+		samplesCondition := &status.Condition{
+			Type:   aro.SamplesOperatorModified,
+			Status: v1.ConditionFalse,
+			Reason: "RedHatKeyNotPresent",
+		}
+
+		// parse keys and validate JSON
+		parsedKeys, err := r.parseRHRegistryKeys(ps)
```

Comment:
name do not correspond to the purpose. It does secret unmarshaling/Verify. Maybe
`r.parsePullSecret` ?